staging/src/k8s.io/client-go/tools/clientcmd/api/v1/conversion.go

```diff
@@ -167,6 +168,9 @@ func addConversionFuncs(scheme *runtime.Scheme) error {
 				if err := s.Convert(&curr.Context, newContext, 0); err != nil {
 					return err
 				}
+				if len(curr.Name) == 0 {
```

Comment:
conversion isn't the right place to do validation like this... validate at point of use

Reply:
For such `conversionFuncs` registered/initialized at `init`, it is quite hard to do the validate at point of use. Will looks quite ugly if added IMO.

Reply:
Errors in conversion do not surface well. This data should not prevent conversion.

Reply:
Validate here: https://github.com/dixudx/kubernetes/blob/20da5a0b5f46f10054ea4d3eacaaf760a890cc1b/staging/src/k8s.io/client-go/tools/clientcmd/validation.go